swiftide-integrations/src/openai/mod.rs

```diff
@@ -308,4 +309,96 @@ mod test {
             Some("gpt-3".to_string())
         );
     }
+
+    #[test]
+    fn test_context_length_exceeded_error() {
+        // Create an API error with the context_length_exceeded code
+        let api_error = ApiError {
+            message: "This model's maximum context length is 8192 tokens".to_string(),
+            r#type: Some("invalid_request_error".to_string()),
+            param: Some("messages".to_string()),
+            code: Some("context_length_exceeded".to_string()),
+        };
+
+        let openai_error = OpenAIError::ApiError(api_error);
+        let result = openai_error_to_language_model_error(openai_error);
+
+        // Verify it's categorized as ContextLengthExceeded
+        match result {
+            LanguageModelError::ContextLengthExceeded(_) => {} // Expected
+            _ => panic!("Expected ContextLengthExceeded error, got {:?}", result),
+        }
+    }
+
+    #[test]
+    fn test_api_error_permanent() {
+        // Create a generic API error (not context length exceeded)
+        let api_error = ApiError {
+            message: "Invalid API key".to_string(),
+            r#type: Some("invalid_request_error".to_string()),
+            param: Some("api_key".to_string()),
+            code: Some("invalid_api_key".to_string()),
+        };
+
+        let openai_error = OpenAIError::ApiError(api_error);
+        let result = openai_error_to_language_model_error(openai_error);
+
+        // Verify it's categorized as PermanentError
+        match result {
+            LanguageModelError::PermanentError(_) => {} // Expected
+            _ => panic!("Expected PermanentError, got {:?}", result),
+        }
+    }
+
+    #[test]
+    fn test_file_save_error_is_permanent() {
+        // Create a file save error
+        let openai_error = OpenAIError::FileSaveError("Failed to save file".to_string());
+        let result = openai_error_to_language_model_error(openai_error);
+
+        // Verify it's categorized as PermanentError
+        match result {
+            LanguageModelError::PermanentError(_) => {} // Expected
+            _ => panic!("Expected PermanentError, got {:?}", result),
```

Comment:
```rust
assert_matches!(result, LanguageModelError::PermanentError(..))
```

https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html

Reply:
nice maar dat is een nightly feature blijkbaar

Reply:
Oh oops, you can still use `matches!` though, the above syntax shouldn't be necessary ever.